### PR TITLE
fix: extract github actions secrets

### PR DESCRIPTION
**Pull request by githinho**
Extract `INFURA_API_KEY` and `ETHERSCAN_API_KEY` to GitHub action secrets.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +110,14 @@ make deploy
 Before deploying, update the constructor-args variable within the Makefile to include any parameters applicable. Make sure to seperate each argument only by a space, no commas.
 
 The deploy script is coded to deploy the "Strategy" contract within the Strategy.sol file. This can be updated by simply updating to src/YourContract.sol:YourContract.
+
+## GitHub Actions
+
+This template comes with GitHub Actions pre-configured. Your contracts will be linted and tested on every push to
+`master` and `develop` branch.
+
+Note though that to make this work, you must se your `INFURA_API_KEY` and your `ETHERSCAN_API_KEY` as GitHub secrets.
```
Comment (storming0x) [inline README.md:119]: small typo*

**Review by storming0x — CHANGES_REQUESTED** [review]
small typo and should be good to merge

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -23,6 +23,6 @@ jobs:
 
       - name: Run tests
         env:
-          ETH_RPC_URL: https://mainnet.infura.io/v3/b7821200399e4be2b4e5dbdf06fbe85b
-          ETHERSCAN_API_KEY: 6776WKCX4XHV7C6NQBTIME6XKM2YNBK29V
+          ETH_RPC_URL: https://mainnet.infura.io/v3/${{ secrets.INFURA_API_KEY }}
```
Comment (storming0x) [inline .github/workflows/test.yaml:26]: I added the SECRETS , but tests CI doesn't work, CI step in trace seems to not find the etherscan key

Comment (storming0x) [inline .github/workflows/test.yaml:26]: Added secrets to environment and repo secrets but makefile step doesnt seem to pull them for some reason

> Run make trace
> forge test -vvv --fork-url https://mainnet.infura.io/v3/  --etherscan-api-key 
> error: The argument '--etherscan-api-key <ETHERSCAN_KEY>' requires a value but none was supplied

**Review by storming0x — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yaml (hunk 2 above)
Comment (githinho) [reply .github/workflows/test.yaml:26]: `${{ secrets.INFURA_API_KEY }}` is the syntax just for Github actions(CI).  This PR will apply only as CI changes.
If you run it locally, you should have `.env` file the same as before. I have `.env` file locally and `make trace` works correctly. Let me know if still have the same problem.

The setup for actions secrets should look like this:
image

Comment (storming0x) [reply .github/workflows/test.yaml:26]: > The setup for actions secrets should look like this: image

Understood, i added the secrets to repo as the image shows, still the CI step is not passing, check the fail message below the PR,

i was looking into this example shared: https://gist.github.com/sambacha/89129cdeb59fd6098a5b46ea8b9664e2

Maybe replacing here:

https://github.com/storming0x/foundry_strategy_mix/blob/8342b1f11a04954c573caa76de65fce06c5cf537/.github/workflows/test.yaml#L28

`forge test -vvv --fork-url "$ETH_RPC_URL" --etherscan-api-key "$ETHERSCAN_API_KEY"`

Comment (githinho) [reply .github/workflows/test.yaml]: I have changed the command for running test, no more `make`. Also, added ETH_RPC_URL secret on my side and it works.
image

**Review by storming0x — APPROVED** [review]
will merge in case for some reason

**Pull request merged**